lib/Ohmbrewer_RIMS.cpp

```diff
@@ -358,8 +404,23 @@ int Ohmbrewer::RIMS::doUpdate(String &args, Ohmbrewer::Equipment::args_map_t &ar
  * @returns The list of physical pins that the Equipment is connected to.
  */
 std::list<int>* Ohmbrewer::RIMS::whichPins() const {
-    // TODO: Implement RIMS::whichPins
+    std::list<int>* pins = new std::list<int>();
+    pins->push_back(_tube->getSensor()->getBusPin());
+    pins->push_back(_tube->getElement()->getPowerPin());
+    pins->push_back(_tunSensor->getBusPin());
+    pins->push_back(_recirc->getPowerPin());
+    pins->push_back(_recirc->getControlPin());
+//    std::list<int> pins = {(_tube->getSensor()->getBusPin()), (_tube->getElement()->getPowerPin()), (_tunSensor->getBusPin()),
+//    (_recirc->getPowerPin()), (_recirc->getControlPin())};;
+
+//    pins = _tube->whichPins();
+//    pins->push_back(&(_tunSensor->getBusPin()));
+//    pins->push_back(&(_recirc->getPowerPin()));
+//    pins->push_back(&(_recirc->getControlPin()));
+
+    // [ tube temp busPin ; tube heating powerPin ; tube heating controlPin ; mashtun temp busPin ; pump powerPin ; pump controlPin]
+
     //return {&_pins, &getTube()->whichPins(), &getTunSensor()->whichPins(), &getRecirculator()->whichPins()};
-    return _pins;
+    return pins;
```

Comment:
Instead of manually specifying the pins for the tube and recirculation pump, just concatenate the list to what you've already pushed into pins.
See http://stackoverflow.com/questions/1449703/how-to-append-a-listt-object-to-another for details.

i.e. 
```
tubePins = _tube->whichPins();
recircPins = _recirc->whichPins();
pins->insert(pins.end(), tubePins.begin(), tubePins.end());
//etc.
```

The advantage here is that you don't have to keep specifying the order (consistently, every time), and if for some strange reason the pin variables, etc. for the tube were to change, you don't have to update RIMS.